app/src/main/java/org/openobservatory/ooniprobe/common/TestDescriptorManager.kt

```diff
@@ -21,4 +29,30 @@ class TestDescriptorManager @Inject constructor(private val context: Context) {
     fun getTestByDescriptorName(name: String): DynamicTestSuite? {
         return getDescriptorByName(name)?.getTest(context)
     }
+
+    /**
+     * Fetches the descriptor from the ooni server using the run id.
+     * @param runId the run id of the descriptor to fetch
+     * @param context the context to use for the request
+     */
+    fun fetchDescriptorFromRunId(runId: Long, context: Context): OONIRunDescriptor {
+        val session = EngineProvider.get().newSession(
+            EngineProvider.get().getDefaultSessionConfig(
+                context,
+                BuildConfig.SOFTWARE_NAME,
+                BuildConfig.VERSION_NAME,
+                LoggerArray(),
+                (context.applicationContext as Application).preferenceManager.proxyURL
+            )
+        )
+        val ooniContext = session.newContextWithTimeout(300)
+
+        val response: OONIRunFetchResponse = session.ooniRunFetch(ooniContext, runId)
+        return response.descriptor
+    }
+
+    fun addDescriptor(descriptor: OONIRunDescriptor, automatedUpdates: Boolean): Boolean {
+        // persist to database
```

Comment:
Is this a TODO? If so, maybe

```suggestion
        // TODO(aanorbel): persist to database
```